src/ngraph/op/broadcast.cpp

```diff
@@ -15,29 +15,278 @@
+    {
+        if (input(0).get_partial_shape().is_static() &&
+            input_value(1).get_node_shared_ptr()->is_constant())
+        {
+            auto arg_shape = input(0).get_shape();
+            auto target_shape =
+                static_pointer_cast<op::Constant>(input_value(1).get_node_shared_ptr())
+                    ->get_shape_val();
+            auto start_axis = (m_broadcast_spec.m_type == AutoBroadcastType::PDPD)
+                                  ? m_broadcast_spec.m_axis
+                                  : target_shape.size() - arg_shape.size();
+            NGRAPH_CHECK(start_axis >= 0);
+            for (auto i = 0; i < target_shape.size(); i++)
+            {
+                if (i < start_axis || target_shape[i] != arg_shape[i - start_axis])
+                {
+                    broadcast_axes.insert(i);
+                }
+            }
+            axes_known = true;
+        }
+    }
+    else
+    {
+        throw ngraph_error("Unknown autobroadcast type");
+    }
+
+    return std::make_pair(axes_known, broadcast_axes);
+}
+
+void op::v1::Broadcast::validate_and_infer_types()
+{
+    // shape node should have integer data type. For now we only allow i64
+    auto shape_et = get_input_element_type(1);
+    NODE_VALIDATION_CHECK(this,
+                          shape_et.compatible(element::Type_t::i64),
+                          "Broadcast shape must have element type i64, but has ",
+                          shape_et);
+
+    // shape node should produce a one dimensional shape.
+    auto broadcast_shape_rank = get_input_partial_shape(1).rank();
+    NODE_VALIDATION_CHECK(this,
+                          broadcast_shape_rank.compatible(1),
+                          "Broadcast shape rank must be 1, but has ",
+                          broadcast_shape_rank);
+
+    if (m_broadcast_spec.m_type == AutoBroadcastType::NONE)
+    {
+        // axes_mapping node should have integer data type. For now we only allow i64
+        auto axes_et = get_input_element_type(2);
+        NODE_VALIDATION_CHECK(this,
+                              axes_et.compatible(element::Type_t::i64),
+                              "Broadcast axes must have element type i64, but has ",
+                              axes_et);
+
+        // axes_mapping node should produce a one dimensional shape.
+        auto axes_shape_rank = get_input_partial_shape(2).rank();
+        NODE_VALIDATION_CHECK(this,
+                              axes_shape_rank.compatible(1),
+                              "Broadcast axes rank must be 1, but has ",
+                              axes_shape_rank);
+    }
+
+    PartialShape result_shape{PartialShape::dynamic()};
+    if (input_value(1).get_node_shared_ptr()->is_constant())
+    {
+        result_shape = static_pointer_cast<op::Constant>(input_value(1).get_node_shared_ptr())
+                           ->get_shape_val();
+    }
+
+    if (m_broadcast_spec.m_type == AutoBroadcastType::NONE)
```

Comment:
According to spec there are only two types of broadcasting NUMPY and EXPLICIT, so what is NONE and PDPD?

Reply:
NONE is the same as EXPLICIT. AutoBroadcastSpec is also being used in elementwise with implicit broadcasting. So I left the name as 'NONE'. i can create an alias if that helps. 
PDPD is paddle paddle auto-broadcast. The definition is in util/attr_types.hpp. We cant assume that numpy is the only possibility here.

Reply:
I agree with NONE for me it's looks more natural than EXPLICIT. I just want spec to be aligned with operations, so in this case I would suggest us to update spec with NONE. Also if PDPD has different broadcast logic it should be also described in spec.

Reply:
@jbobba, should we update the spec with this new broadcasting type? PDPD? Can you please talk with @slyalin, or please create a merge request to the spec.